# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../nail/openssl-1.0.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

rediffed for our version of nail after
http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/mail-client/nail/files/nail-12.4-openssl-1.patch?revision=1.1

--- mailx-12.3/openssl.c.orig	2006-03-04 01:01:19.000000000 +0100
+++ mailx-12.3/openssl.c	2011-04-28 13:38:54.148000254 +0200
@@ -105,7 +105,7 @@
 static void ssl_load_verifications(struct sock *sp);
 static void ssl_certificate(struct sock *sp, const char *uhp);
 static enum okay ssl_check_host(const char *server, struct sock *sp);
-static int smime_verify(struct message *m, int n, STACK *chain,
+static int smime_verify(struct message *m, int n, STACK_OF(X509) *chain,
 		X509_STORE *store);
 static EVP_CIPHER *smime_cipher(const char *name);
 static int ssl_password_cb(char *buf, int size, int rwflag, void *userdata);
@@ -308,7 +308,7 @@
 	X509 *cert;
 	X509_NAME *subj;
 	char data[256];
-	/*GENERAL_NAMES*/STACK	*gens;
+	/*GENERAL_NAMES*/STACK_OF(GENERAL_NAMES)	*gens;
 	GENERAL_NAME	*gen;
 	int	i;
 
@@ -494,7 +494,7 @@
 }
 
 static int
-smime_verify(struct message *m, int n, STACK *chain, X509_STORE *store)
+smime_verify(struct message *m, int n, STACK_OF(X509) *chain, X509_STORE *store)
 {
 	struct message	*x;
 	char	*cp, *sender, *to, *cc, *cnttype;
@@ -503,7 +503,8 @@
 	off_t	size;
 	BIO	*fb, *pb;
 	PKCS7	*pkcs7;
-	STACK	*certs, *gens;
+	STACK_OF(X509)	*certs;
+	STACK_OF(GENERAL_NAMES) *gens;
 	X509	*cert;
 	X509_NAME	*subj;
 	char	data[LINESIZE];
@@ -612,7 +613,7 @@
 {
 	int	*msgvec = vp, *ip;
 	int	ec = 0;
-	STACK	*chain = NULL;
+	STACK_OF(X509)	*chain = NULL;
 	X509_STORE	*store;
 	char	*ca_dir, *ca_file;
 
@@ -685,7 +686,7 @@
 	X509	*cert;
 	PKCS7	*pkcs7;
 	BIO	*bb, *yb;
-	STACK	*certs;
+	STACK_OF(X509)	*certs;
 	EVP_CIPHER	*cipher;
 
 	certfile = expand((char *)certfile);
@@ -948,9 +949,9 @@
 	off_t	size;
 	BIO	*fb, *pb;
 	PKCS7	*pkcs7;
-	STACK	*certs;
+	STACK_OF(X509)	*certs;
 	X509	*cert;
-	STACK	*chain = NULL;
+	STACK_OF(X509)	*chain = NULL;
 	enum okay	ok = OKAY;
 
 	message_number = n;
